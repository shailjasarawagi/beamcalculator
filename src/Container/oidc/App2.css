.App2 {
    text-align: center;
    padding-top:100px;
  }
  
  .App2-logo {
    animation: NewApp-logo-spin infinite 20s linear;
    height: 40vmin;
    pointer-events: none;
  }
  
  .App2-header {
    background-color: #282c34;
    min-height: 100vh;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    font-size: calc(10px + 2vmin);
    color: white;
  }
  
  .App2-link {
    color: #61dafb;
  }
  
  @keyframes NewApp-logo-spin {
    from {
      transform: rotate(0deg);
    }
    to {
      transform: rotate(360deg);
    }
  }
  
