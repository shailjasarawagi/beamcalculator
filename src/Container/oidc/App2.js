/**
 * This manages the redirection of url when user is not authenticated
 */
import React, { Component } from 'react';
import classes from './App2.css';
//import axios from '../../hoc/axios-orders';
import * as functions from '../../Functions/Function';
import { withRouter } from 'react-router-dom';

/**send_request() is called when user is not authenticated and a new window is opened */
class App2 extends Component {

  componentDidMount() {
    this.status = "KO";
    this.pollInterval = null;
    this.polling_id = null;
    this.send_request();
  }

  state = {
    redirect_url: null
  }
  
  /**This opens a new window when user not authenticated */
  send_request = () => {
    // console.log("request sent")
    // axios.get('/oidc/auth/request/', {})
    //   .then(response => {
    //     // console.log("hello response", response.data);
    //     // this.setState({
    //     //   redirect_url: "df"
    //     // })
    //     // this.pooling_id = response.data.pooling_id;
    //     this.window = window.open(response.data.redirect_url, "_self");
    //     // localStorage.setItem("pollingId", this.pooling_id);
    //   })
    //   .catch((error) => {
    //     console.log("error", error);
    //   });

        // const url=process.env.REACT_APP_BACKEND_URL;
        const url=functions.getBackendDomain().baseURL;

      window.open(url+"/oidc/auth/request/", "_self");
  }

 
  render() {
    return (
      <div className={classes.App2}>
      </div>
    )
  }
}

export default withRouter(App2)