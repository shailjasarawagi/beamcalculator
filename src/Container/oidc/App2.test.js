import React from "react";
import { configure, shallow, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import App2 from "../oidc/App2";
import { cleanup } from '@testing-library/react';
// import { MemoryRouter } from 'react-router-dom'
// import MockAdapter from 'axios-mock-adapter';
import axios from "axios";


afterEach(cleanup);

configure({ adapter: new Adapter() });



describe('Layout', () => {
    it('should render the component without errors', () => {
        const wrapper = mount(
            <App2.WrappedComponent />
        );
        expect(wrapper).toMatchSnapshot();
    });
    describe('App2', () => {
        test('async test', async () => {
            const asyncMock = jest.fn().mockResolvedValue();
            await asyncMock();

        });
        it('success', async () => {
            const asyncMock = jest.fn().mockResolvedValue();
            const spyOn = jest.spyOn(console, 'log');
            asyncMock(<App2.WrappedComponent />);
            await
                expect(spyOn).toHaveBeenCalledTimes(0);
            spyOn.mockRestore();
        });

    });


    it('a post request should be made', () => {
        const wrapper = mount(
            <App2.WrappedComponent />
        );
        const postSpy = jest.spyOn(axios, 'get');
        // console.log(postSpy);
    });


    // it('accepts a callback that returns an axios request', function () {
    //     const wrapper = mount(
    //         <MemoryRouter initialEntries={[{ pathname: '/' }]}>
    //             <App2 />
    //         </MemoryRouter>,
    //     );
    //     var instance;
    //     var mock = new MockAdapter(instance);
    //     mock
    //         .onGet('/oidc/auth/request/')
    //         .reply(404, { data: 'aaa' })


    //     return axios.get('/oidc/auth/request/').then(function (response) {
    //         expect(response.status).toEqual(404);

    //     });
    // });
});







