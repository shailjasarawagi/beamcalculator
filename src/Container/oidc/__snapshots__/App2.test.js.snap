// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should render the component without errors 1`] = `ReactWrapper {}`;
